workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/RegistryValidator.java

```diff
@@ -42,10 +42,19 @@
 	
 	private Set<String> registries=new HashSet<String>();
 
+
+	/**
+	 * 1. It will check that spec has buildspec or dockerfile 
+	 * 2. If both is not then it will return true
+	 * 3. If any one is there then 
+	 *    3.1  It will fetch registry details
+	 *    3.2  Then it will verify the registry
+	 *    3.3  if registry is exist which is provided by user then return true else false
+	 */
 	@Override
 	public boolean validate(WorkflowContext context) throws HyscaleException {
 		logger.debug("Starting registry validation");
-		if (!ImageDetailsUtil.isImageBuildPushRequired(context.getServiceSpec())) {
+		if (!ImageUtil.isImageBuildPushRequired(context.getServiceSpec())) {
 			return true;
 		}
 		ImageRegistry imageRegistry = context.getServiceSpec().get(
```

Comment:
This needs to be fixed as discussed on the call. Here we only need the registry from Image Object.

Reply:
Registry is required for 2 things:
  1. For pushing to target registry ( only incase of dockerfile or buildSpec is provided )
  2. Creating an ImagePullSecret (In service image case, image creds may or may not be present )
         - Leave a warn message if you don't find the credentials for a service image.